pkg/sm/sm.go

```diff
@@ -110,23 +110,31 @@ func New(ctx context.Context, cancel context.CancelFunc, env env.Environment) *S
 	log.C(ctx).Info("Setting up Service Manager storage...")
 
 	waitGroup := &sync.WaitGroup{}
-	smStorage := &postgres.PostgresStorage{}
-	if err := storage.OpenWithSafeTermination(ctx, smStorage, cfg.Storage, waitGroup); err != nil {
-		panic(fmt.Sprintf("error opening storage: %s", err))
+	encrypter := &security.AESEncrypter{}
+
+	smStorage := &postgres.Storage{
+		ConnectFunc: func(driver string, url string) (*sql.DB, error) {
+			return sql.Open(driver, url)
+		},
+		Encrypter: encrypter,
 	}
 
-	securityStorage := smStorage.Security()
-	if err = initializeSecureStorage(ctx, securityStorage); err != nil {
-		panic(fmt.Sprintf("error initialzing secure storage: %v", err))
+	// Initialize the storage with graceful termination
+	if err := storage.InitializeWithSafeTermination(ctx, smStorage, cfg.Storage, waitGroup); err != nil {
+		panic(fmt.Sprintf("error opening storage: %s", err))
 	}
 
-	encrypter := &security.TwoLayerEncrypter{
-		Fetcher: securityStorage.Fetcher(),
+	// Decorate the storage with credentials ecryption/decryption
+	encryptingRepository, err := storage.NewEncryptingRepository(ctx, smStorage, encrypter)
+	if err != nil {
+		panic(fmt.Sprintf("error setting up encrypting repository: %s", err))
 	}
 
+	// Decorate the storage with transactional interceptors
+	interceptableRepository := storage.NewInterceptableTransactionalRepository(encryptingRepository)
```

Comment:
I'd prefer all of this decoration to be hidden behind a configuration

Reply:
wouldnt it be too hidden then ? Is your idea to pass a config to SM builder that contains decorators or do you mean something else?

Reply:
That would be an option `smb.EnableStorageInterceptors().EnableStorageEncryption()`. Or just the storage configuration can be extended with options to allow interceptors and encryption. 
```
storage:
    uri: ....
    interceptors: true
    encryption: true
```